services/data/postgres_async_db.py

```diff
@@ -162,7 +162,8 @@ async def _init(self, create_triggers: bool):
             await PostgresUtils.setup_trigger_notify(db=self.db, table_name=self.table_name, keys=self.trigger_keys)
 
     async def get_records(self, filter_dict={}, fetch_single=False,
-                          ordering: List[str] = None, limit: int = 0, expanded=False) -> DBResponse:
+                          ordering: List[str] = None, limit: int = 0, expanded=False,
+                          cur: aiopg.Cursor = None) -> DBResponse:
```

Comment:
Cursor lifecycle is closely tied to DB transaction lifecycle

For tag mutation, we want to use the same cursor when executing logic that is dispersed across multiple existing DB access functions (read run, do some checks, write run tag).  The idea is to pass in your own cursor as an optional in these existing DB access functions.  If we pass in the cursor, the inner DB code will use that (without managing its lifetime, like "closing" it), instead of always creating a new one.